src/ensembl/src/content/app/entity-viewer/gene-view/GeneView.tsx

```diff
@@ -191,6 +192,8 @@ const GeneViewWithData = (props: GeneViewWithDataProps) => {
       return;
     }
 
+    dispatch(closeSidebarModal());
```

Comment:
No, I disagree.

The `closeSidebarModal` function should be called at the GeneView level in the following two cases:

1) When user changes species/genome
2) When user switches to a different "app"

In order to do this,

1) We should keep track of the active genome id from the previous render of the component
2) If during the current render the genome id changes, close the sidebar
3) If the GeneView component gets unmounted, close the sidebar

in order for this to happen:

```tsx
// in the imports
import usePrevious from 'src/shared/hooks/usePrevious';

// in the component, keep track of the genome id during previous render:

const previousGenomeId = usePrevious(genomeId);

// also in the component, create a new useEffect

useEffect(() => {
  if (genomeId !== previousGenomeId) {
    dispatch(closeSidebarModal());
  }

  return () => dispatch(closeSidebarModal());
}, [genomeId]);
```

Reply:
1. We are closing the sidebar modal upon Geneview unmount. So do we need an extra comparison with the previous genome id?
2. If the genome id changes, that means GeneView is unmounted/re-rendered which calls the useEffect and closes the sidebar modal

Reply:
Unmount is different from rerender. We are currently not running any code when GeneView unmounts.

But I re-read the useEffect and agree that it will work as we intend it to. Only instead of closing the modal when GeneView unmounts, it will close the modal every time GeneView mounts.

I guess we can keep it as it is then.

Reply:
> Only instead of closing the modal when GeneView unmounts, it will close the modal every time GeneView mounts.
I didnt get this. Isnt that the same?

Reply:
It will behave similarly, but the underlying mechanism will be different. With one mechanism, the component cleans up after itself, with the other it cleans up after the previous component when it arrives 😄